### PR TITLE
Eslint for feed-discovery and fix lints

### PR DESCRIPTION
**Pull request by JiaHua-Zou**
## Issue This PR Addresses
#3161 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Setting up the `Eslint` for `feed-discovery` and fix some lint error.
Lint errors:
1. Dependencies not listed in package.json:
Fix: adding the dependencies to the `devDependencies` list. 
2. calling the `new` operator without assigning to a variable:
 Fix: Created a function that will return the URL if the URL is valid, if not throws an error.

## Steps to test the PR
pnpm install
pnpm lint

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by AmasiaNalbandian** [comment]
Hello! Please rebase, as the following commit https://github.com/Seneca-CDOT/telescope/commit/6046273fba4b89ce6281f7a34cafbdec7ef222dc has reorganized our file structure!

**Comment by JiaHua-Zou** [comment]
Pr has to rebase

File: src/api/feed-discovery/src/middleware.js [hunk 1]
```diff
@@ -2,16 +2,25 @@ const got = require('got');
 const cheerio = require('cheerio');
 const { logger, createError } = require('@senecacdot/satellite');
 
+// check if the URL is valid.
+const isUrl = (url) => {
+  try {
+    return new URL(url);
+  } catch {
+    throw createError(400, 'Invalid Blog URL');
```
Comment (tcvan0707) [inline src/api/feed-discovery/src/middleware.js:10]: Why don't you return `false` here, then throwing `400, 'Invalid Blog URL'` on line 23?

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by DukeManh — CHANGES_REQUESTED** [review]

**Review by humphd — DISMISSED** [review]
Looks good, but I'd like to see a clean CI run with these changes.

**Review by tcvan0707 — DISMISSED** [review]

**Review by cindyledev — APPROVED** [review]

**Review by cindyledev — APPROVED** [review]

**Review by Kevan-Y — APPROVED** [review]

**Pull request merged**